include/geode/geosciences_io/model/helpers/geos_export.h

```diff
@@ -0,0 +1,497 @@
+
+                    const auto model_property =
+                        model_.block( polygon_mesh_element.mesh_id )
+                            .mesh()
+                            .polyhedron_attribute_manager()
+                            .template find_attribute< double >( property_name );
+                    auto value = model_property->value(
+                        polygon_mesh_element.element_id );
+                    solid_property->set_value( polyhedron_id, value );
+                }
+            }
+            for( const auto& property_name : cell_2Dproperty_names_ )
+            {
+                auto solid_property =
+                    model_solid_->polyhedron_attribute_manager()
+                        .template find_or_create_attribute<
+                            geode::VariableAttribute, std::array< double, 2 > >(
+                            property_name, { 0., 0. } );
+                for( const auto polyhedron_id :
+                    geode::Range( model_solid_->nb_polyhedra() ) )
+                {
+                    auto polygon_mesh_element =
+                        brep_mesh_elements->value( polyhedron_id );
+
+                    const auto model_property =
+                        model_.block( polygon_mesh_element.mesh_id )
+                            .mesh()
+                            .polyhedron_attribute_manager()
+                            .template find_attribute< std::array< double, 2 > >(
+                                property_name );
+                    auto value = model_property->value(
+                        polygon_mesh_element.element_id );
+                    solid_property->set_value( polyhedron_id, value );
+                }
+            }
+            for( const auto& property_name : cell_3Dproperty_names_ )
+            {
+                auto solid_property =
+                    model_solid_->polyhedron_attribute_manager()
+                        .template find_or_create_attribute<
+                            geode::VariableAttribute, std::array< double, 3 > >(
+                            property_name, { 0., 0., 0. } );
+                for( const auto polyhedron_id :
+                    geode::Range( model_solid_->nb_polyhedra() ) )
+                {
+                    auto polygon_mesh_element =
+                        brep_mesh_elements->value( polyhedron_id );
+
+                    const auto model_property =
+                        model_.block( polygon_mesh_element.mesh_id )
+                            .mesh()
+                            .polyhedron_attribute_manager()
+                            .template find_attribute< std::array< double, 3 > >(
+                                property_name );
+                    auto value = model_property->value(
+                        polygon_mesh_element.element_id );
+                    solid_property->set_value( polyhedron_id, value );
+                }
+            }
```

Comment:
Do you think these three similar code blocks could be merged into one (templated?) function?

Reply:
I tryied to do something generic evaluating the type of attributes ect. but i did not succeed if you know how to do it. Please let me know.